platform/pensando/sonic-platform-modules-dpu/dpu/utils/fetch_dpu_status

```diff
@@ -0,0 +1,338 @@
+            pcielink_stat['reason'] = 'pdsagent is not running on dpu container'
+            pcielink_stat['timestamp'] = datetime.now().strftime('%Y%m%d %H:%M:%S')
+            return pcielink_stat
+
+        reason = ''
+        try:
+            cmd = "/nic/bin/pdsctl show lif | grep -i adminstate"
+            output = apiHelper.run_docker_cmd(cmd)
+            fields = output.split()
+            admin_state_index = fields.index('AdminState') - 1
+            oper_state_index = fields.index('OperState') - 1
+            cmd = "/nic/bin/pdsctl show lif | grep INTERNAL-MGMT"
+            output = apiHelper.run_docker_cmd(cmd)
+            int_mgmt_data = output.split()
+            admin_status = int_mgmt_data[admin_state_index]
+            oper_status = int_mgmt_data[oper_state_index]
+            if admin_status == 'UP' and oper_status == 'UP':
+                int_mgmt_status = True
+            else:
+                int_mgmt_status = False
+            reason += f'INTERNAL-MGMT : admin state - {admin_status}, oper_state - {oper_status},'
+            reason += f' status - {bool_to_healthd_status(int_mgmt_status)}, '
+
+            cmd = "/nic/bin/pdsctl show lif | grep HOST-MGMT"
+            output = apiHelper.run_docker_cmd(cmd)
+            host_mgmt_data = output.split()
+            admin_status = host_mgmt_data[admin_state_index]
+            oper_status = host_mgmt_data[oper_state_index]
+            if admin_status == 'UP' and oper_status == 'UP':
+                host_mgmt_status = True
+            else:
+                host_mgmt_status = False
+            reason += f'HOST-MGMT : admin state - {admin_status}, oper_state - {oper_status},'
+            reason += f' status - {bool_to_healthd_status(host_mgmt_status)}'
+
+            state = "down"
+            if host_mgmt_status and int_mgmt_status:
+                print('dpu-pcie_link:OK')
+                state = "up"
+            else:
+                print('dpu-pcie_link:Not OK')
+
+            pcielink_stat = {}
+            pcielink_stat['state'] = state
+            pcielink_stat['reason'] = reason
+            pcielink_stat['timestamp'] = datetime.now().strftime('%Y%m%d %H:%M:%S')
+            return pcielink_stat
+        except:
+            print('dpu-pcie_link:Not OK')
+            return {}
+
+    def _refresh_dpu_health_status(self):
+        try:
+            pdsagent_name = self._fetch_pdsagent_status()
+            self._fetch_pciemgrd_status()
+            eth_link_stats = self._fetch_eth_link_status()
+            pcie_link_stats = self._fetch_pcie_link_status()
+
+            fvs_data = [
+                ('id', self.slot_id),
+                ('dpu_midplane_link_state', pcie_link_stats.get('state', NOT_AVAILABLE)),
+                ('dpu_midplane_link_time', pcie_link_stats.get('timestamp', NOT_AVAILABLE)),
+                ('dpu_midplane_link_reason', pcie_link_stats.get('reason', NOT_AVAILABLE)),
+            ]
```

Comment:
@shanshri @rameshraghupathy midplane up/down can be monitored by the NPU and we have NPU chasisd monitoring the midplane

Reply:
@prgeor Though the NPU can monitor the midplane failure and update, if there are other types of midplane related issues/errors when the midplane is reachable, that can't be detected by NPU chassisd, can be reported from the DPU.